Sources/Purchasing/OfferingsManager.swift

```diff
@@ -24,42 +24,77 @@ class OfferingsManager {
     private let backend: Backend
     private let offeringsFactory: OfferingsFactory
     private let productsManager: ProductsManagerType
+    private let diagnosticsTracker: DiagnosticsTrackerType?
+    private let dateProvider: DateProvider
 
     init(deviceCache: DeviceCache,
          operationDispatcher: OperationDispatcher,
          systemInfo: SystemInfo,
          backend: Backend,
          offeringsFactory: OfferingsFactory,
-         productsManager: ProductsManagerType) {
+         productsManager: ProductsManagerType,
+         diagnosticsTracker: DiagnosticsTrackerType?,
+         dateProvider: DateProvider = DateProvider()) {
         self.deviceCache = deviceCache
         self.operationDispatcher = operationDispatcher
         self.systemInfo = systemInfo
         self.backend = backend
         self.offeringsFactory = offeringsFactory
         self.productsManager = productsManager
+        self.diagnosticsTracker = diagnosticsTracker
+        self.dateProvider = dateProvider
     }
 
     func offerings(
         appUserID: String,
         fetchPolicy: FetchPolicy = .default,
         fetchCurrent: Bool = false,
+        trackDiagnostics: Bool = true,
         completion: (@MainActor @Sendable (Result<Offerings, Error>) -> Void)?
     ) {
-        guard !fetchCurrent && !self.systemInfo.dangerousSettings.uiPreviewMode else {
-            self.fetchFromNetwork(appUserID: appUserID, fetchPolicy: fetchPolicy, completion: completion)
-            return
-        }
+        self.trackPurchaseStartedIfNeeded(trackDiagnostics: trackDiagnostics)
+        let startTime = self.dateProvider.now()
 
-        guard let memoryCachedOfferings = self.cachedOfferings else {
-            self.fetchFromNetwork(appUserID: appUserID, fetchPolicy: fetchPolicy, completion: completion)
-            return
-        }
+        self.systemInfo.isApplicationBackgrounded { isAppBackgrounded in
```

Comment:
This will now cause a change in threads when fetching offerings... But that may also happen at any point if fetching offerings from network. So I think this is ok? It does mean we know check whether the app is backgrounded or not before fetching offerings in order to calculate the cache status, which might have some very slight behavior changes.

Reply:
I see that the code is now executed inside a `Task { @MainActor `, which means that all this code now happens on the main thread. It shouldn't have any consequences, I think.

Perhaps the `dispatchCompletionOnMainThreadIfPossible` below could be simplified to call the `completion`, but since the completion in `isApplicationBackgrounded` is not marked with  `@MainActor`, it's safer this way (keeping `dispatchCompletionOnMainThreadIfPossible`).